src/Widgets/MultitaskingView.vala

```diff
@@ -544,28 +547,48 @@ namespace Gala {
          * {@inheritDoc}
          */
         public void open (HashTable<string,Variant>? hints = null) {
-            if (opened)
-                return;
+            bool manual_animation = hints != null && hints.get ("manual_animation").get_boolean ();
+
+            if (!opened) {
+                if (manual_animation) {
+                    debug ("Starting MultitaskingView manual open animation");
+                    this.gesture_animation_director = new GestureAnimationDirector();
```

Comment:
I'm not sure why is there a need to create a new instance of this class everytime open/close is called here? If the intention is to manage some kind of state, then perhaps it would be better to initialize the director only once and then store the needed flag in it?